sbc_compassion/migrations/12.0.1.0.4/post-migration.py

```diff
@@ -0,0 +1,51 @@
+from openupgradelib import openupgrade
+
+
+def _update_reading_language(cr, sponsorship_id, lang):
+    cr.execute(
+        "UPDATE recurring_contract "
+        "SET reading_language = %s "
+        "WHERE id = %s", [lang, sponsorship_id]
+    )
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    if not version:
+        return
+
+    sponsorships_to_fill = env["recurring.contract"].search([
+        ("reading_language", "=", False),
+        ("child_id", "!=", False),
+        ("state", "not in", ["terminated", "cancelled"])
+    ])
+
+    message_obj = env["gmc.message"]
+    action_id = env.ref("sponsorship_compassion.create_sponsorship").id
+
+    for sponsorship in sponsorships_to_fill:
+        correspondent_lang_id = False
+        for lang in sponsorship.correspondent_id.spoken_lang_ids:
+            if sponsorship.child_id.correspondence_language_id.id == lang.id:
+                _update_reading_language(
+                    env.cr,
+                    sponsorship.id,
+                    lang.id
+                )
+                continue
```

Comment:
In this case you may not store the `correspondent_lang_id` value and you won't create the message update. So maybe you should add here `correspondent_lang_id = lang.id` before the `continue` statement.

Reply:
Checking for the existence of the variable `reading_language` should also do the trick without changing the rest of the code !